src/connectors/tls_connector.rs

```diff
@@ -58,30 +62,35 @@ impl<C> TlsConnector<C> {
 
     #[cfg(not(feature = "native-tls"))]
     #[inline]
-    pub fn new_with_tls_default(inner_connector: C) -> Self {
+    pub fn new_with_tls_default(inner_connector: C, alpn: Option<Vec<Vec<u8>>>) -> Self {
```

Comment:
Should ALPN be a property of connector or passed for each connect?

Reply:
Yes, it should be a property of the connector, since the connector/client will advertise the protocols it supports for every new connection it initiates.